src/condor_collector.V6/view_server.cpp

```diff
@@ -586,34 +586,34 @@ ViewServer::findOffset(FILE* & /*fp*/, int FromDate, int ToDate, ExtIntArray* ti
 			return NULL;
 		}
 
-	} else if( times_array->getElementAt( times_array->getlast() ) < FromDate ) {
+	} else if( times_array->back() < FromDate ) {
 
 			// linear progression, but start with the latest known offset
 		// dprintf(D_ALWAYS, "Requesting time at the end of where the current cache knows\n");
-		search_offset_ptr = offsets->getElementAt( offsets->getlast() );
+		search_offset_ptr = offsets->back();
 
 	} else {
 
 			// binary search through times_array[] for the latest time not
 			// greater than FromDate
 		// dprintf(D_ALWAYS, "Binary searching the cache table, request should be quick\n");
 		int low = 0;
-		int high = times_array->getlast();
+		int high = times_array->back();
```

Comment:
getlast() returns the index of the last element in the array, not a reference to it.
```suggestion
		int high = times_array->size()-1;
```